contributors/devel/kubemark-guide.md

```diff
@@ -43,62 +51,83 @@ Common workflow for Kubemark is:
+  3. Creates a VM instance for the master, configured with the PD and IP created above.
+  4. Set firewall rule in the master to open port 443\* for all TCP traffic by default.
 
 <sub>\* Port 443 is a secured port on the master machine which is used for all
 external communication with the API server. In the last sentence *external*
 means all traffic coming from other machines, including all the Nodes, not only
 from outside of the cluster. Currently local components, i.e. ControllerManager
 and Scheduler talk with API server using insecure port 8080.</sub>
 
+- [Optional to read] Establish necessary certs/keys required for setting up the PKI for kubemark cluster:
+  (*the steps below are independent of GCE and work for all providers*)
+  1. Generate a randomly named temporary directory for storing PKI certs/keys which is delete-trapped on EXIT.
+  2. Create a bearer token for 'admin' in master.
+  3. Generate certificate for CA and (certificate + private-key) pair for each of master, kubelet and kubecfg.
+  4. Generate kubelet and kubeproxy tokens for master.
+  5. Write a kubeconfig locally to `test/kubemark/kubeconfig.kubemark` for enabling local kubectl use.
+
+- Set up environment and start master components (through `start-kubemark-master.sh` script):
+  (*the steps below use gcloud for SSH and SCP to master, and should be easy to do outside of GCE*)
+  1. SSH to the master machine and create a new directory (`/etc/srv/kubernetes`) and write all the
+     certs/keys/tokens/passwords to it.
+  2. SCP all the master pod manifests, shell scripts (`start-kubemark-master.sh`, `configure-kubectl.sh`, etc),
+     config files for passing env variables (`kubemark-master-env.sh`) from the local machine to the master.
+  3. SSH to the master machine and run the startup script `start-kubemark-master.sh` (and possibly others).
+
+  Note: The directory structure and the functions performed by the startup script(s) can vary based on master distro.
+        We currently support the GCI image `gci-dev-56-8977-0-0` in GCE.
+
+- Set up and start HollowNodes (as pods) on the external cluster:
+  (*the steps below (except 2nd and 3rd) are independent of GCE and work for all providers*)
+  1. Identify the right kubemark binary from the current kubernetes repo for the platform linux/amd64.
+  2. Create a Docker image for HollowNode using this binary and upload it to a remote Docker repository.
+     (We use gcr.io/ as our remote docker repository in GCE, should be different for other providers)
+  3. [One-off] Create and upload a Docker image for NodeProblemDetector (see kubernetes/node-problem-detector repo),
+     which is one of the containers in the HollowNode pod, besides HollowKubelet and HollowProxy. However we
+     use it with a hollow config that esentially has an empty set of rules and conditions to be detected.
+     (Doing this outside of gcr.io might get slightly tricky involving tweaking with the n-p-d repo)
```

Comment:
Made the wording of this statement better.
I was referring to pushing the docker image to a non-GCR registry in case of other cloud providers.
As they would have to change the makefile (and there are a few issues in it with the go flags currently) and run `make` to upload the image at the desired place.